We will be using reads for two samples of Arabidopsis (Columbia the reference genome and Seattle a wilder variety)
For each sample, we have 2 fastq files: R1 and R2. Given the size of the files, we couldn't upload them here. Bellow instructions and URL to access files.

Access to **Seattle samples**: <br/>
Download sample 1:
```
wget http://www.cgr.liv.ac.uk/illum/BecAmodule_2244855b7b53c241/Mapping-exercise-Seattle-reads/seattle0_r1_filter.fastq
```
Download sample 2:
```
wget http://www.cgr.liv.ac.uk/illum/BecAmodule_2244855b7b53c241/Mapping-exercise-Seattle-reads/seattle0_r2_filter.fastq
```
Access to **Columbia samples**:<br/>
Download sample 1:
```
wget http://www.cgr.liv.ac.uk/illum/BecAmodule_2244855b7b53c241/Mapping-exercise-Columbia-reads/Columbia_R1_filter.fastq
```
Download sample 2:
```
wget http://www.cgr.liv.ac.uk/illum/BecAmodule_2244855b7b53c241/Mapping-exercise-Columbia-reads/Columbia_R2_filter.fastq
```

Download the **reference genome** fasta file: 
```
wget http://www.cgr.liv.ac.uk/illum/BecAmodule_2244855b7b53c241/Mapping-exercise-Reference-sequence/Arabidopsis_TAIR10_chromosome1.fasta
```
